blockchain/proxy_test.go

```diff
@@ -111,6 +129,13 @@ func createMockArgsProxy(httpClient sdkHttp.Client) ArgsProxy {
 	}
 }
 
+func createMockArgsProxyWithCache(httpClient sdkHttp.Client, cacher BlockDataCache) ArgsProxy {
+	args := createMockArgsProxy(httpClient)
```

Comment:
createMockArgsProxy should have the DisabledCache and you can change the cacher without this function